src/presentation/ui/component/form/passwordInput.templ

```diff
@@ -18,60 +19,141 @@ script PasswordInputLocalState() {
+						<i class="ph-duotone ph-eye-slash" x-show="!isPasswordReadable"></i>
+						<i class="ph-duotone ph-eye" x-show="isPasswordReadable"></i>
+					</button>
+				</div>
+				<label
+					class="from-os-300 via-os-300 absolute -top-2 left-1.5 z-[1] cursor-text bg-gradient-to-t via-50% to-transparent to-50% px-1.5 text-xs font-bold text-neutral-50 text-opacity-80 transition-all before:absolute before:left-0 before:top-0 before:z-[-1] before:block before:h-full before:w-full before:bg-white before:transition-all peer-placeholder-shown:top-2.5 peer-placeholder-shown:text-sm peer-placeholder-shown:font-normal peer-autofill:-top-2 peer-hover:text-opacity-90 peer-focus:-top-2 peer-focus:cursor-default peer-focus:text-opacity-100"
+				>
+					{ label }
+				</label>
+			</div>
+			<div class="group relative">
+				<button
+					type="button"
+					@click={ bindModelPath + " = generateRandomPassword()" }
+					class="inline-flex items-center rounded-md border border-cyan-800 bg-cyan-950 px-3 py-2 lowercase hover:border-transparent hover:bg-cyan-800"
+				>
+					<i class="ph-bold ph-binary text-xl"></i>
 				</button>
+				<div class={ "absolute opacity-0 bottom-full mb-2 transition-all duration-300 group-hover:block group-hover:-translate-y-0.5 group-hover:opacity-100 w-max p-2 bg-cyan-900 text-white text-xs rounded-md shadow-lg" }>
+					generate random password
+				</div>
 			</div>
-			<label
-				for={ id }
-				class="from-os-300 via-os-300 absolute -top-2 left-1.5 z-[1] cursor-text bg-gradient-to-t via-50% to-transparent to-50% px-1.5 text-xs font-bold text-neutral-50 text-opacity-80 transition-all before:absolute before:left-0 before:top-0 before:z-[-1] before:block before:h-full before:w-full before:bg-white before:transition-all peer-placeholder-shown:top-2.5 peer-placeholder-shown:text-sm peer-placeholder-shown:font-normal peer-autofill:-top-2 peer-hover:text-opacity-90 peer-focus:-top-2 peer-focus:cursor-default peer-focus:text-opacity-100"
-			>
-				{ label }
-			</label>
 		</div>
-		<div class="relative group">
-			<button
-				type="button"
-				@click={ bindValuePath + " = generateRandomPassword()" }
-				class="inline-flex items-center rounded-md px-3 py-2 lowercase bg-cyan-950 hover:bg-cyan-800 border border-cyan-800 hover:border-transparent"
-			>
-				<i class="ph-bold ph-binary text-xl"></i>
-			</button>
-			<div class={ "absolute opacity-0 bottom-full mb-2 transition-all duration-300 group-hover:block group-hover:-translate-y-0.5 group-hover:opacity-100 w-max p-2 bg-cyan-900 text-white text-xs rounded-md shadow-lg" }>
-				generate random password
-			</div>
+		<!-- PasswordInputStrengthProgressBar HTML -->
+		<div class="bg-os-200 mb-4 h-1.5 rounded-full bg-opacity-70">
+			<div x-bind:class="'bg-infinite-500 h-1.5 rounded-full w-' + strengthPercentage + '%'"></div>
+		</div>
+		<!-- PasswordInputStrengthCriteria HTML -->
+		<div class="w-full">
+			<ul x-show="shouldShowPasswordStrengthCriteria && strengthPercentage > 0">
```

Comment:
You may rename shouldShowPasswordStrengthCriteria to just "displayPassStrengthCriteria".

Reply:
I chose to keep "Password" written out instead "Pass" to maintain the same pattern used in the component state names.